zephyr/include/sof/schedule/task.h

```diff
@@ -101,6 +101,6 @@ void task_main_init(void);
 
 void task_main_free(void);
 
-int task_main_start(struct sof *sof);
+int task_main_start(void);
 
```

Comment:
The best way to handle this is to switch from ``struct sof *`` to ``void *`` because in the end even zephyr will need a context of execution.

Reply:
As I can see it's still doing a sof_get() to get to the sof context. So it's still using the sof context.

Reply:
The idea is that sof_get() is used inside SOF module, the Zephyr does not know about it. Otherwise I need to include headers and there is a header include chain conflicting with Zephyr headers.

Reply:
@finikorg I'm still not convinced that this is the best solution, but since you merge this into the zephyr branch it is OK for now. We can find a better solution later.